frontend/src/components/personal/PreferencesPage.tsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+/* eslint-disable no-magic-numbers */
+
+import React, { useState, useCallback, useEffect } from 'react'
+import {useDispatch, useSelector} from 'react-redux'
+import {Card, CardActions, CardContent, CardHeader, CardMedia, Button, 
+  Typography, Avatar, Box, TextField } from '@material-ui/core'
+
+import {changeMyData, getMyData} from '../../actions/prefAndProfileActions'
+import {useStyles} from './ProfilePreferencesPage.styles'
+import {RootState} from '../../reducers/index'
+import './PreferencesPage.scss'
+
+export default function PreferencesPage (): React.ReactElement {
+  const dispatch = useDispatch()
+  
+  useEffect(() => {
+        dispatch(getMyData())
+    }, [dispatch])  
+
+  // @ts-ignore
```

Comment:
Please, work on this `// @ts-ignore` to remove it. You will need to specify return type of your reducer.